### PR TITLE
[Feat] EVM prefill email

### PR DESCRIPTION
**Pull request by q20274982**
## Summary
[SDK] EVM prefill email

## Related Links

**Asana**:  https://app.asana.com/0/1201812548509877/1204201697295380/f

**Mockup**:  

## Checklist

- [ v] Pasted Asana or Mockup link.
- [ ] Tagged labels.

### CODE REVIEW

File: src/providers/ethereum.ts [hunk 1]
```diff
@@ -153,7 +153,12 @@ export default class EthereumProvider extends BloctoProvider implements Ethereum
     }
 
     if (!this.connected) {
-      await this.enable();
+      let email: string | undefined;
```
Comment (yanglin5689446) [inline src/providers/ethereum.ts:156]: I'd prefer `let email?: string`

Comment (q20274982) [reply src/providers/ethereum.ts]: [[Refactor] refactor with URLSearchParams & regex](https://github.com/portto/blocto-sdk/pull/56/commits/aa31cb01a211394ebf07321f82875436539ddb7e)

File: src/providers/ethereum.ts [hunk 2]
```diff
@@ -252,7 +252,7 @@ export default class EthereumProvider extends BloctoProvider implements Ethereum
       }
 
       const location = encodeURIComponent(window.location.origin);
-      const loginFrame = createFrame(`${this.server}/${this.appId}/${this.chain}/authn?l6n=${location}`);
+      const loginFrame = createFrame(`${this.server}/${this.appId}/${this.chain}/authn?l6n=${location}${email && `&email=${email}`}`);
```
[the remark this answers is not in the document]
Comment (q20274982) [reply src/providers/ethereum.ts]: [[Refactor] refactor with URLSearchParams & regex](https://github.com/portto/blocto-sdk/pull/56/commits/aa31cb01a211394ebf07321f82875436539ddb7e)

**Review by yanglin5689446 — APPROVED** [review]

**Review by yanglin5689446 — APPROVED** [review]
LGTM 👍

**Pull request merged**